src/commands/deployment.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the ""License"");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an ""AS IS"" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { Listr, type ListrTaskWrapper } from 'listr2'
+import { SoloError } from '../core/errors.js'
+import { BaseCommand } from './base.js'
+import * as flags from './flags.js'
+import { constants, Templates } from '../core/index.js'
+import * as prompts from './prompts.js'
+import chalk from 'chalk'
+import { RemoteConfigTasks } from '../core/config/remote/remote_config_tasks.js'
+import { ListrLease } from '../core/lease/listr_lease.js'
+import type { Namespace } from '../core/config/remote/types.js'
+import type { CommandFlag, ContextClusterStructure } from '../types/index.js'
+
+export class DeploymentCommand extends BaseCommand {
+  private static get DEPLOY_FLAGS_LIST (): CommandFlag[] {
+    return [
+      flags.quiet,
+      flags.namespace,
+      flags.deploymentName,
+      flags.userEmailAddress,
+      flags.deploymentClusters,
```

Comment:
you give the option to supply `--deployment-clusters`, but when I use it, it still looks for `--context-cluster`.   There is some more work that Ivo is doing in LocalConfig in his current PR that should help us get the context when you already have the cluster.  I'm wondering if we want to use that.  Or, if you want to just remove deploymentClusters?   Not sure what the purpose would be otherwise.

```
❯ cat ~/.solo/cache/local-config.yaml
userEmailAddress: jeromy@swirldslabs.com
deployments:
  solo-e2e:
    clusters:
      - kind-solo-e2e
currentDeploymentName: solo-e2e
❯ k9s -A
❯ ls -al ~/.solo/logsq
❯ solo deployment create -n solo-e2e --deployment-clusters kind-solo-e2e

******************************* Solo *********************************************
Version			: 0.99.0
Kubernetes Context	: kind-solo-e2e
Kubernetes Cluster	: kind-solo-e2e
Kubernetes Namespace	: solo-e2e
**********************************************************************************
(node:63822) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
✖ Cannot read properties of undefined (reading 'split')
◼ Prompt local configuration
◼ Validate cluster connections
◼ Create remote config
*********************************** ERROR *****************************************
Error installing chart solo-deployment
***********************************************************************************
```